live-build/misc/ansible-roles/appliance-build.virtualization-common/tasks/main.yml

```diff
@@ -45,3 +45,49 @@
       domain=
       dns1=
       dns2=
+
+#
+# Initial NFS Server configuration for virtualization. We do this here instead
+# of within the virtualization package because the virtualization software
+# modifies this configuration at runtime, and we don't want ansible roles that
+# are re-run as part of upgrades to undo these runtime modifications.
+#
+- lineinfile:
+    path: /etc/default/nfs-kernel-server
+    regexp: "^{{ item.key }}="
+    line: "{{ item.key }}=\"{{ item.value }}\""
+    insertafter: EOF
+    state: present
+  with_items:
+      #
+      # Increase nfsd threads beyond default of 8
+      #
+      - { key: "RPCNFSDCOUNT", value: "64" }
+      #
+      # Drop --manage-gids flag for nfs-mountd service
+      #
+      - { key: "RPCMOUNTDOPTS", value: "--no-nfs-version 2" }
+      #
+      # We ship out of the box with an NFSv4-only configuration to limit the
+      # security surface area of the product for the common case. The
+      # virtualization product can re-enable NFSv3 at runtime as needed. We
+      # also disable 4.1 and 4.2 (thus limiting the server to 4.0) in order to
+      # work around performance issues found with those minor versions.
+      #
+      - { key: "RPCNFSDOPTS", value: "-d --no-nfs-version 2 --no-nfs-version 3 --no-nfs-version 4.1 --no-nfs-version 4.2" }
+
+#
+# Because we want an NFSv4-only configuration out of the box, we need to mask
+# NFSv3 services so that they don't get automatically started at boot via
+# dependencies. The virtualization software is responsible for unmasking and
+# starting these services if NFSv3 needed at runtime.
+#
+- name: Mask NFSv3 services
+  file:
+    src: "/dev/null"
+    dest: "/etc/systemd/system/{{ item }}"
+    state: link
+  with_items:
+      - nfs-mountd.service
```

Comment:
Can we move this to the `debian/postinst` file, just to be consistent with where we manipulate other services?

Also, can we not call `systemctl mask` in the build/chroot environment?

Reply:
I think putting this in `debian/postinst` of any package would result in our potentially losing our customizations during in-place upgrades, right?

We might be able to call systemctl, I'll have to try it.

Reply:
> I think putting this in debian/postinst of any package would result in our potentially losing our customizations during in-place upgrades, right?

Ah yes.. Sorry, I mistakenly read this as logic that'll run on upgrade too (i.e. in delphix-platform); ignore my comment.

> Also, can we not call systemctl mask in the build/chroot environment?

I doubt that we can use the `systemd` ansible module, so I'm good with your solution. I think it's better than using `command: systemctl ...` in ansible (`command` isn't idempotent, whereas your solution is).